In [26]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats

performance = pd.read_csv(r'C:\Users\casey\Downloads\product-performance.csv')
sales = pd.read_csv(r'C:\Users\casey\Downloads\sales-data.csv')

print(performance.head())
#print("\n---------------\n")
print(sales.head())

      Category  Revenue  Profit_Margin  Marketing_Spend  Units_Sold  \
0      Laptops  2450000           28.5           520000        4250   
1  Smartphones  1850000           32.0           410000        7200   
2      Tablets   980000           30.5           220000        3900   
3  Accessories   720000           45.0           150000       24000   
4    Wearables   640000           38.0           140000        5300   

   Customer_Satisfaction   ROI  Inventory_Turnover  
0                    4.2  4.71                 5.8  
1                    4.3  4.51                 7.2  
2                    4.1  4.45                 5.3  
3                    4.4  4.80                 8.5  
4                    4.5  4.57                 6.7  
  Month  Year   Sales  Marketing_Spend  Transactions  Online_Sales  \
0   Jan  2023  425000            85000          3250        297500   
1   Feb  2023  410000            80000          3100        287000   
2   Mar  2023  475000            95000       

# Task 1 Sales Performance Analysis Using Hypothesis Testing



In [20]:
sales_2024 = sales[sales['Year'] == 2024]
sales_2024_mean = sales_2024['Sales'].mean()
target = 650000

#Compares the mean of the sample to a target value to find if it is significantly different
t_statistic, p_value = stats.ttest_1samp(sales_2024['Sales'], target)

print("One-Sample T-Test Results:")
print(f"Sample mean: {sales_2024['Sales'].mean():.2f}")
print(f"Target value: {target}")
print(f"T-statistic: {t_statistic:.4f}")
print(f"P-value: {p_value:.4f}")

# interprets the results
if p_value < 0.05:
    print("Result: SIGNIFICANT - Mean differs from benchmark")
else:
    print("Result: NOT SIGNIFICANT - No evidence the mean differs from benchmark")


One-Sample T-Test Results:
Sample mean: 632916.67
Target value: 650000
T-statistic: -0.4641
P-value: 0.6516
Result: NOT SIGNIFICANT - No evidence the mean differs from benchmark


The p value is greatly above `0.05` which shows us that the difference is insignificant. The t statistic is also relatively insignificant, as it's incredibly close to `0`. The sales are close to the benchmark.

In [ ]:
# Two sample t test
sales_2023 = sales[sales['Year'] == 2023]
sales_2023_sales = sales_2023['Sales'] #  monthly sales for 2023
sales_2024_sales = sales_2024['Sales']

t_statistic, p_value = stats.ttest_ind(sales_2023_sales, sales_2024_sales)

print("Two-Sample T-Test Results:")
print(f"Campaign A mean: ${sales_2023_sales.mean():.2f}")
print(f"Campaign B mean: ${sales_2024_sales.mean():.2f}")
print(f"Difference: ${sales_2024_sales.mean() - sales_2023_sales.mean():.2f}")
print(f"T-statistic: {t_statistic:.4f}")
print(f"P-value: {p_value:.4f}")

# If the p value is < 0.05 then we can confidently say that the 2024 sales performed better than 2023
if p_value < 0.05:
    print("Result: SIGNIFICANT")
    if sales_2024_sales.mean() > sales_2023_sales.mean():
        print("Sales 2024 performs better!")
else:
    print("Result: NOT SIGNIFICANT")

Two-Sample T-Test Results:
Campaign A mean: $580000.00
Campaign B mean: $632916.67
Difference: $52916.67
T-statistic: -1.0890
P-value: 0.2879
Result: NOT SIGNIFICANT


There is an observable difference in sales mean, but the p value `0.2879` being > `0.05` shows us that this difference could be realistically explained by random variation. The T-Statistic is also pretty insignificant.

From a business perspective, sales have improved, but not significantly, we can't promise that the sales have improved in a way that can't be explained by random variation. If there was an a new promotion or increase in advertising spending, these results might show us that the campaign was somewhat ineffective. 

In [34]:
# task 2: product performance evaluation 
profit_margin = performance['Profit_Margin'] / 100 # converting to percentage
benchmark = 0.30

# one sample t test
t_stat, p_value = stats.ttest_1samp(profit_margin, benchmark)
print(f"T-Statistic: {t_stat:.2f}")
print(f"P-value: {p_value:.2f}")
print(f"Profit margin: {profit_margin.mean():.2f}")

if p_value < 0.05:
    print("There is a significant difference from the benchmark")
else:
    print("There is not a significant difference from the benchmark")


T-Statistic: 0.31
P-value: 0.76
Profit margin: 0.31
There is not a significant difference from the benchmark


The p value `0.76` is significantly greater than 0.05, which means that the results were not significant. However, there are two possible interpretations of the result being significant. We could either be way above or way below the benchmark, and a different strategy would have to be suited to that outcome. In either case, it would probably come down the efficiency of our pricing strategy. Either we are exceeding expectations due to effective pricing or we are failing to reach our desired profit due to a bad pricing strategy.